src/ast/semantic_analyser.cpp

```diff
@@ -879,20 +879,47 @@ void SemanticAnalyser::visit(Call &call)
   }
   else if (call.func == "time") {
     check_assignment(call, false, false, false);
-    if (check_varargs(call, 0, 1)) {
-      if (is_final_pass()) {
-        if (call.vargs && call.vargs->size() > 0) {
+    if (check_varargs(call, 0, 2))
+    {
+      if (is_final_pass())
+      {
+        if (call.vargs == nullptr || call.vargs->size() == 0)
+        {
+          std::string fmt_default = "%H:%M:%S\n";
+          bpftrace_.time_args_.push_back(fmt_default.c_str());
+          warning("Using `time()` to print time directly is now deprecated.",
```

Comment:
"directly" is a bit confusing in this context. Maybe this is more clear: "Using `time()` to implicitly print time is deprecated. Print the returned value instead".